lib/matplotlib/backends/backend_gtk3.py

```diff
@@ -411,7 +411,9 @@ def destroy(self, *args):
     def show(self):
         # show the figure window
         self.window.show()
-        self.window.present()
+        self.canvas.draw()
```

Comment:
Why does GTK3 need a draw now?

Reply:
When I am testing with Gtk3 and turn off the "raise window", I encountered a strange behavior, that the GUI window is reluctant to update, and the whole program get stuck, loop is not running, until I move/switch the GUI window or moving my mouse cursor on the canvas, it can continue to run/update.
After debug and tracing, I found the program is stuck inside the `Gtk.main_iteration()` here:
```python
    def flush_events(self):
        # docstring inherited
        Gdk.threads_enter()
        while Gtk.events_pending():
            Gtk.main_iteration()
        Gdk.flush()
        Gdk.threads_leave()
```
By looking into the doc, it says if there is no event queued, `main_iteration` will block until the next event is noticed, but that's very strange since we have already checked with `events_pending`, if there is no event it should not go into the while-loop. Nevertheless, I changed using `Gtk.gtk_main_iteration_do(False)` which the doc states that it won't block, however, that's not working, the program still get stuck inside `gtk_main_iteration_do`. Finally, I come up with a solution: to call `window.queue_draw()` right after `window.show()` and successfully I don't have to waving my mouse cursor over the canvas any more. And I found that this is exactly what `canvas.draw()` is doing, and the same approach is also used in the `backend_wx.py`, so I suppose this is the right way to do it? I am not familiar with the GUI stuff so I have no idea what's happening, please feel free to correct me.
This is the full report, HaHa!! Thank you so much for your patience reading this mumbo jumbo!!  :stuck_out_tongue_winking_eye:

Reply:
Could you test with `draw_idle` instead?

Reply:
tested, won't update, need stick to `draw`